src/io/vss_store.rs

```diff
@@ -0,0 +1,201 @@
+use io::Error;
+use std::io;
+use std::io::ErrorKind;
+#[cfg(test)]
+use std::panic::RefUnwindSafe;
+
+use crate::io::utils::check_namespace_key_validity;
+use lightning::util::persist::KVStore;
+use tokio::runtime::Runtime;
+use vss_client::client::VssClient;
+use vss_client::error::VssError;
+use vss_client::types::{
+	DeleteObjectRequest, GetObjectRequest, KeyValue, ListKeyVersionsRequest, PutObjectRequest,
+};
+
+/// A [`KVStore`] implementation that writes to and reads from a [VSS](https://github.com/lightningdevkit/vss-server/blob/main/README.md) backend.
+pub struct VssStore {
+	client: VssClient,
+	store_id: String,
+	runtime: Runtime,
+}
+
+impl VssStore {
+	#[cfg(any(vss, vss_test))]
+	pub(crate) fn new(base_url: &str, store_id: String) -> Self {
+		let client = VssClient::new(base_url);
+		let runtime = tokio::runtime::Builder::new_multi_thread().enable_all().build().unwrap();
+		Self { client, store_id, runtime }
+	}
+
+	fn build_key(
+		&self, primary_namespace: &str, secondary_namespace: &str, key: &str,
+	) -> io::Result<String> {
+		if primary_namespace.is_empty() {
+			Ok(key.to_string())
+		} else {
+			Ok(format!("{}#{}#{}", primary_namespace, secondary_namespace, key))
```

Comment:
Should we use `/` for the separator so that error messages don't have two different keys when the server returns one in its error message?

Reply:
FWIW, those are front-slashes, which aren't use for escaping. 😄  But makes sense if the backend may be using a filesystem instead of database, I suppose.